tiledb/sm/misc/parallel_functions.h

```diff
@@ -270,7 +270,9 @@ Status parallel_for(
   }
 
   // Wait for all instances of `execute_subrange` to complete.
-  throw_if_not_ok(tp->wait_all(tasks));
+  // This is ignoring the wait status as we use failed_exception for propagating
+  // the tasks exceptions.
+  std::ignore = tp->wait_all(tasks);
```

Comment:
The earlier change was better. Other recent changes now have `parallel_for` able to throw. There's no reason not to keep moving forward on this.

Reply:
This was causing the following test failure only for azure linux_asan check.  

```
/home/vsts/work/1/s/test/src/unit-filter-pipeline.cc:4008: FAILED:
  {Unknown expression after the reported line}
due to a fatal error condition:
  SIGABRT - Abort (abnormal termination) signal
```

I think it was caused because some of the thread exceptions were not caught, causing SIGABRT.

Reply:
> /home/vsts/work/1/s/test/src/unit-filter-pipeline.cc:4008: FAILED:
> I think it was caused because some of the thread exceptions were not caught, causing SIGABRT.

Line 4008:
>    CHECK(tile.alloc_data(nelts * sizeof(uint64_t)).ok());

Looking inside that function, I'd rather guess that the `assert` is failing. `alloc_data` is being called earlier in the function, which would violate the assumption that `alloc_data` is checking.

Reply:
Used `throw_if_not_ok ` instead of `RETURN_NOT_OK`